### PR TITLE
Add command-line option to display the progress indicator

### PR DESCRIPTION
**Pull request by lkitching**
Issue #132 - Add --progress-bar command-line flag to indicate a
progress bar should be displayed. The progress bar writes a '.' to
stderr every two seconds during processing.

### CODE REVIEW

File: src/csv2rdf/main.clj [hunk 1]
```diff
@@ -49,6 +52,30 @@
...
+        pf (.scheduleAtFixedRate e write-progress 2 2 TimeUnit/SECONDS)]
+    (reify AutoCloseable
+      (close [_]
+        (.cancel pf false)
+        (.shutdown e)))))
+
+(defn- get-progress-writer
+  "Starts a progress bar according to the requested options. Returns an AutoCloseable
+   instance that stops the progress bar on close."
+  [{:keys [progress-bar] :as opts}]
```
[the remark this answers is not in the document]
Comment (Robsteranium) [reply src/csv2rdf/main.clj:73]: Although it's unused I think this helps explain the nature of the argument.

File: src/csv2rdf/main.clj [hunk 2]
```diff
@@ -75,10 +102,11 @@
               :rdf-format (or (some-> output-file formats/->rdf-format) RDFFormat/TURTLE)
               :mode mode}
         output-file (some-> output-file io/file)]
-    (if output-file
-      (with-open [w (io/writer output-file)]
-        (write-output w opts))
-      (write-output (io/writer *out*) opts))))
+    (with-open [progress-monitor (get-progress-writer options)]
```
[the remark this answers is not in the document]
Comment (Robsteranium) [reply src/csv2rdf/main.clj:105]: It's not used in the body but that's fine as it's on another thread anyway.

**Comment by Robsteranium** [comment]
Hello, we need this for ons-geo-graft. [The tag we were depending on](https://github.com/Swirrl/csv2rdf/commit/3db25) seems to have fallen from the tree, presumably because it's superseded by this PR.

Can we merge it @lkitching @jennet?

**Comment by Robsteranium** [comment]
Is the need for this obviated by [travis_wait](https://github.com/Swirrl/ons-geo-graft/blob/master/.travis.yml#L38)?

**Comment by jennet** [comment]
> Hello, we need this for ons-geo-graft. [The tag we were depending on](https://github.com/Swirrl/csv2rdf/commit/3db25) seems to have fallen from the tree, presumably because it's superseded by this PR.
> 
> Can we merge it @lkitching @jennet?

@RickMoynihan suggested running the csv2rdf command in combination with [pipe viewer](http://www.ivarch.com/programs/pv.shtml), which was enough for my needs on [ecology](https://github.com/Swirrl/defra-ecology/blob/main/data-pipeline/biosys-ref.makefile#L11) (see also [the vars file](https://github.com/Swirrl/defra-ecology/blob/main/data-pipeline/vars.mk#L7)), would this be any use to you?

**Pull request closed without merging**